packages/block-library/src/paragraph/edit.js

```diff
@@ -176,12 +128,33 @@ function ParagraphBlock( {
 
 	const richTextStyles = useMemo(
 		() => ( {
-			backgroundColor: backgroundColor.color,
-			color: textColor.color,
 			fontSize: fontSize.size ? fontSize.size + 'px' : undefined,
 			direction,
 		} ),
-		[ backgroundColor.color, textColor.color, fontSize.size, direction ]
+		[ fontSize.size, direction ]
+	);
+
+	const { TextColor, BackgroundColor, InspectorControlsColorPanel } = __experimentalUseColors(
+		[
+			{ name: 'textColor', property: 'color' },
+			{ name: 'backgroundColor', property: 'background-color', className: 'has-background' },
+		],
+		{
+			panelChildren: ( components ) => {
+				return (
+					<ContrastChecker
+						{ ...{
+							fallbackTextColor,
+							fallbackBackgroundColor,
+							fontSize,
+						} }
+						textColor={ components.TextColor.color }
+						backgroundColor={ components.BackgroundColor.color }
+					/>
+				);
+			},
```

Comment:
Can we use the simpler `contrastCheckerProps` instead?

Reply:
Hi @epiqueras, It seemed not to fit this use case because:
- The default contrast mechanism is to check the contrast for all the colors, which means that the backgroundColor itself would be checked and so we would have a warning all the time.
- The backgroundColor for contrast checking passed to hook depends on one of the colors the hook returns, which we don't have yet available when we pass it.